frontend/windows/handle_zip.go

```diff
@@ -0,0 +1,188 @@
+package windows
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/moby/buildkit/client/llb"
+	"github.com/moby/buildkit/exporter/containerimage/image"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	workerImgRef    = "mcr.microsoft.com/mirror/docker/library/ubuntu:jammy"
+	outputDir       = "/tmp/output"
+	buildScriptName = "_build.sh"
+)
+
+var (
+	varCacheAptMount = llb.AddMount("/var/cache/apt", llb.Scratch(), llb.AsPersistentCacheDir("dalec-windows-var-cache-apt", llb.CacheMountLocked))
+	varLibAptMount   = llb.AddMount("/var/lib/apt", llb.Scratch(), llb.AsPersistentCacheDir("dalec-windows-var-lib-apt", llb.CacheMountLocked))
+)
+
+func handleZip(ctx context.Context, client gwclient.Client, spec *dalec.Spec) (gwclient.Reference, *image.Image, error) {
+	sOpt, err := frontend.SourceOptFromClient(ctx, client)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pg := dalec.ProgressGroup("Build windows container: " + spec.Name)
+	worker := workerImg(sOpt, pg)
+
+	bin, err := buildBinaries(spec, worker, sOpt)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to build binaries: %w", err)
+	}
+
+	st := getZipLLB(worker, spec.Name, bin)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	def, err := st.Marshal(ctx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error marshalling llb: %w", err)
+	}
+
+	res, err := client.Solve(ctx, gwclient.SolveRequest{
+		Definition: def.ToPB(),
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	ref, err := res.SingleRef()
+	// Do not return a nil image, it may cause a panic
+	return ref, &image.Image{}, err
+}
+
+type RunOptFunc func(*llb.ExecInfo)
```

Comment:
This should move to the root dalec package or be unexported.

Reply:
I removed this in the latest commit since I found something better.

Reply:
done